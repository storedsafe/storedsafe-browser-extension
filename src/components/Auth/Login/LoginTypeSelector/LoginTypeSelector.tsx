import * as React from 'react';
import { Field } from '../../../Form';
import { LoginType } from '../Login';

export type LoginTypes = {
  title: string;
  value: LoginType;
}[];

export interface LoginTypeSelectorProps {
  loginTypes: LoginTypes;
  value?: string;
  onChange?: (selected: LoginType) => void;
}

export const LoginTypeSelector: React.SFC<LoginTypeSelectorProps> = ({
  loginTypes,
  value,
  onChange,
}: LoginTypeSelectorProps) => (
  <Field
    type="select"
    name="loginType"
    label="Login Type"
    value={value}
    onChange={onChange}
    options={loginTypes} />
);
