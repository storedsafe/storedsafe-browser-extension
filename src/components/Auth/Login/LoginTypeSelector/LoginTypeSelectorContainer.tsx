import * as React from 'react';
import { LoginType } from '../Login';
import { LoginTypeSelector, LoginTypeSelectorProps } from './LoginTypeSelector';

type OnLoginTypeChangeCallback = (loginType: LoginType) => void;

export interface LoginTypeSelectorContainerProps extends LoginTypeSelectorProps {
  onSiteChangeCallback?: OnLoginTypeChangeCallback;
  render: (
    loginType: LoginType,
  ) => React.ReactNode;
}

export const LoginTypeSelectorContainer: React.FC<LoginTypeSelectorContainerProps> = ({
  onSiteChangeCallback,
  render,
  ...props
}: LoginTypeSelectorContainerProps) => {
  const [selected, setSelected] = React.useState<LoginType>(props.loginTypes[0].value);

  const onChange = (selected: LoginType): void => {
    onSiteChangeCallback && onSiteChangeCallback(selected);
    setSelected(selected);
  };

  return (
    <React.Fragment>
      <LoginTypeSelector value={selected} onChange={onChange} {...props} />
      {render(selected)}
    </React.Fragment>
  );
};
