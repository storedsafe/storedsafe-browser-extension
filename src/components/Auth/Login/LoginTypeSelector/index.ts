import { LoginTypeSelectorContainer } from './LoginTypeSelectorContainer';

export default LoginTypeSelectorContainer;
