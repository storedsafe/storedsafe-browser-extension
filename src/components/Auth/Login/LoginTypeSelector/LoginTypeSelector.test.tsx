import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { LoginTypeSelector, LoginTypes } from './LoginTypeSelector';

const loginTypes: LoginTypes = [
  { title: 'YubiKey', value: 'yubikey' },
  { title: 'TOTP', value: 'totp' },
];

test('<LoginTypeSelector />', () => {
  const wrapper = shallow(<LoginTypeSelector loginTypes={loginTypes} />);
  expect(wrapper).toMatchSnapshot();
});

test('<LoginTypeSelector value=totp />', () => {
  const wrapper = shallow(<LoginTypeSelector loginTypes={loginTypes} value={'totp'} />);
  expect(wrapper).toMatchSnapshot();
});

test('<LoginTypeSelector />, integration', () => {
  const onChange = jest.fn();
  const wrapper = mount(<LoginTypeSelector
    loginTypes={loginTypes}
    value={'yubikey'}
    onChange={onChange} />
  );
  const select = wrapper.find('select');
  select.simulate('change', { target: { value: 'yubikey', name: 'loginType' } });
  expect(onChange).toHaveBeenCalledWith('yubikey', 'loginType');
});
