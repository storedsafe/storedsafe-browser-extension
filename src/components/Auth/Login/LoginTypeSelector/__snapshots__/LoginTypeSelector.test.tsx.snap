// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginTypeSelector /> 1`] = `ShallowWrapper {}`;

exports[`<LoginTypeSelector value=totp /> 1`] = `ShallowWrapper {}`;
